### PR TITLE
Delay dropping of root privileges

### PR DESCRIPTION
**Pull request by axkg**
The early dropping of root privileges as introduced with #18 does not work for me, as binding an interface requires root. Without it mdns-repeater reports an error:

`mdns-repeater: send setsockopt(SO_BINDTODEVICE): Operation not permitted`

This PR delays the dropping of the root privileges and moves daemonize() post the opening of the sockets to ensure the pid file is created for the same user that is expected to unlink() it on shutdown.

### CODE REVIEW

File: mdns-repeater.c [hunk 1]
```diff
@@ -79,6 +79,9 @@ int shutdown_flag = 0;
 
 char *pid_file = PIDFILE;
 
+int switch_user_flag = 0;
+const struct passwd* user;
```
Comment (geekman) [inline mdns-repeater.c]: perhaps use `user == NULL` instead of having an additional variable as a flag

File: mdns-repeater.c [hunk 2]
```diff
@@ -558,6 +554,22 @@ int main(int argc, char *argv[]) {
...
+	else {
+		// check for pid file when running in foreground
+		running_pid = already_running();
+		if (running_pid != -1) {
+			log_message(LOG_ERR, "already running as pid %d", running_pid);
+			exit(1);
+		}
+	}
+
+
```
Comment (geekman) [inline mdns-repeater.c]: remove excess empty line

**Review by geekman — CHANGES_REQUESTED** [review]

**Comment by axkg** [comment]
Dropped the flag (and the excess line) as requested, works for me for both cases (with and without -u switch).

**Comment by geekman** [comment]
Alright, looks good. Thanks!

**Pull request merged**